components/search/src/selector.rs

```diff
@@ -317,4 +344,248 @@ mod tests {
+        );
+
+        result = Arc::clone(&selector).filter_engine_configuration(SearchUserEnvironment {
+            locale: "fi".into(),
+            region: "FR".into(),
+            update_channel: SearchUpdateChannel::Default,
+            distribution_id: "starship".to_string(),
+            experiment: String::new(),
+            app_name: SearchApplicationName::Firefox,
+            version: String::new(),
+        });
+
+        assert!(result.is_ok(), "Should not have errored: `{result:?}`");
+        assert_eq!(
+            result.unwrap(),
+            RefinedSearchConfig {
+                engines: vec!(
+                    SearchEngineDefinition {
+                        aliases: Vec::new(),
+                        charset: "UTF-8".to_string(),
+                        classification: SearchEngineClassification::General,
+                        identifier: "test1".to_string(),
+                        name: "Test 1".to_string(),
+                        order_hint: None,
+                        partner_code: String::new(),
+                        telemetry_suffix: None,
+                        urls: SearchEngineUrls {
+                            search: SearchEngineUrl {
+                                base: "https://example.com/1".to_string(),
+                                method: "GET".to_string(),
+                                params: Vec::new(),
+                                search_term_param_name: Some("q".to_string())
+                            },
+                            suggestions: None,
+                            trending: None
+                        }
+                    },
+                    SearchEngineDefinition {
+                        aliases: Vec::new(),
+                        charset: "UTF-8".to_string(),
+                        classification: SearchEngineClassification::General,
+                        identifier: "test3".to_string(),
+                        name: "Test 3".to_string(),
+                        order_hint: None,
+                        partner_code: String::new(),
+                        telemetry_suffix: None,
+                        urls: SearchEngineUrls {
+                            search: SearchEngineUrl {
+                                base: "https://example.com/3".to_string(),
+                                method: "GET".to_string(),
+                                params: Vec::new(),
+                                search_term_param_name: Some("trek".to_string())
+                            },
+                            suggestions: None,
+                            trending: None
+                        }
+                    },
+                ),
+                app_default_engine_id: "test1".to_string(),
+                app_default_private_engine_id: None
+            }, "Should have selected the single engine, as the environments do not match for the other two"
```

Comment:
Could we have more description test statements for this as well? 
Maybe naming the specific properties we're filtering for, "Should have selected engine with distro: starship, and firefox application" ?